<?xml version="1.0"?>
<!--|
    Copyright Inria July 2017
    This project has received funding from the European Research Council (ERC) under
    the European Union's Horizon 2020 research and innovation programme (grant
    agreement No 637972) - see http://www.resibots.eu

    Contributor(s):
      - Matthias Mayr (matthias.mayr@cs.lth.se)
      - Konstantinos Chatzilygeroudis (konstantinos.chatzilygeroudis@inria.fr)
      - Rituraj Kaushik (rituraj.kaushik@inria.fr)
      - Roberto Rama (bertoski@gmail.com)


    This software is governed by the CeCILL-C license under French law and
    abiding by the rules of distribution of free software.  You can  use,
    modify and/ or redistribute the software under the terms of the CeCILL-C
    license as circulated by CEA, CNRS and INRIA at the following URL
    "http://www.cecill.info".

    As a counterpart to the access to the source code and  rights to copy,
    modify and redistribute granted by the license, users are provided only
    with a limited warranty  and the software's author,  the holder of the
    economic rights,  and the successive licensors  have only  limited
    liability.

    In this respect, the user's attention is drawn to the risks associated
    with loading,  using,  modifying and/or developing or reproducing the
    software by the user in light of its specific status of free software,
    that may mean  that it is complicated to manipulate,  and  that  also
    therefore means  that it is reserved for developers  and  experienced
    professionals having in-depth computer knowledge. Users are therefore
    encouraged to load and test the software's suitability as regards their
    requirements in conditions enabling the security of their systems and/or
    data to be ensured and,  more generally, to use and operate it in the
    same conditions as regards security.

    The fact that you are presently reading this means that you have had
    knowledge of the CeCILL-C license and that you accept its terms.

|-->
<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from iiwa_left.urdf.xacro           | -->
<!-- |    But then it was adapted by hand ¯\_(ツ)_/¯                                   | -->
<!-- |    Manual changes:                                                              | -->
<!-- |     - Disabled visuals and collision for the table and the box                  | -->
<!-- |     - Replaced iiwa joint collision mesh with box                               | -->
<!-- |     - Inserted peg                                                              | -->
<!-- |     - Configured safety limits as hardware limits                               | -->
<!-- =================================================================================== -->
<robot name="bh_robot">
  <material name="Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="Green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.4 0.4 0.4 1.0"/>
  </material>
  <material name="Orange">
    <color rgba="1.0 0.423529411765 0.0392156862745 1.0"/>
  </material>
  <material name="Brown">
    <color rgba="0.870588235294 0.811764705882 0.764705882353 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <!-- Insert the BASE -->
  <link name="world"/>
  <!--  Define the origin of your 3D space
  Here we define a box lying at 0,0,0 with some dimensions, in reality our robot is place on top of a metal cart with those dimensions.
  This will include the cart in the scene and avoid the robot to hit it when using MoveIt!
  If you don't want to have that you can define a basic world link (see commented lines)
  -->
  <joint name="footprint_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <joint name="table_origin_joint" type="fixed">
    <parent link="base_link"/>
    <child link="table_origin"/>
    <origin xyz="0 0 0.83"/>
  </joint>
  <joint name="torso_origin_joint" type="fixed">
    <parent link="table_origin"/>
    <child link="torso_origin"/>
    <!-- x: 185mm from the CAD plus 1mm shift in mounting position y: centered half width of the table -->
    <origin xyz="0.186 0.4 0"/>
  </joint>
  <joint name="plate_1_inner_joint" type="fixed">
    <parent link="table_origin"/>
    <child link="plate_1_inner"/>
    <origin rpy="0 0 3.14159265359" xyz="1.45 0.194 0.02"/>
  </joint>
  <!-- commented out since they are typically not used
  <joint name="plate_1_outer_joint" type="fixed">
    <parent link="plate_1_inner"/>
    <child link = "plate_1_outer"/>
    <origin xyz="0 0.05 0" rpy="0 0 0" />
  </joint>
  -->
  <joint name="plate_2_inner_joint" type="fixed">
    <parent link="table_origin"/>
    <child link="plate_2_inner"/>
    <origin rpy="0 0 3.14159265359" xyz="1.45 0.606 0.02"/>
  </joint>
  <!-- commented out since they are typically not used
  <joint name="plate_2_outer_joint" type="fixed">
    <parent link="plate_2_inner"/>
    <child link = "plate_2_outer"/>
    <origin xyz="0 -0.05 0" rpy="0 0 0" />
  </joint>
  -->
  <link name="table_origin">
  </link>
  <link name="torso_origin">
  </link>
  <!-- commented out since they are typically not used
  <link name="plate_1_outer">
  </link>
  -->
  <!-- this one is typically used -->
  <link name="plate_1_inner">
  </link>
  <!-- commented out since they are typically not used
  <link name="plate_2_outer">
  </link>
  -->
  <!-- this one is typically used -->
  <link name="plate_2_inner">
  </link>
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0.8 0.4 0.415"/>
      <mass value="100.0"/>
      <inertia ixx="11.074162237" ixy="0.0" ixz="0.0" iyy="27.074155837" iyz="0.0" izz="26.666656"/>
    </inertial>
    <!-- <visual>
      <origin rpy="0 0 0" xyz="0.8 0.4 0.415"/>
      <geometry>
        <box size="1.6 0.8 0.83"/>
      </geometry>
      <material name="White"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.8 0.4 0.415"/>
      <geometry>
        <box size="1.6 0.8 0.83"/>
      </geometry>
    </collision> -->
  </link>
  <gazebo reference="box_link">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>
  <joint name="box_joint" type="fixed">
    <parent link="base_link"/>
    <child link="box_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <joint name="box_origin_joint" type="fixed">
    <parent link="box_link"/>
    <child link="box_origin"/>
    <origin xyz="-0.01 0 0.567"/>
  </joint>
  <link name="box_link">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.41 0.4 0.2835"/>
      <mass value="100.0"/>
      <inertia ixx="8.01240512837" ixy="0.0" ixz="0.0" iyy="8.01240512837" iyz="0.0" izz="10.6666624"/>
    </inertial>
    <!-- <visual>
      <origin rpy="0 0 0" xyz="-0.41 0.4 0.2835"/>
      <geometry>
        <box size="0.8 0.8 0.567"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.41 0.4 0.2835"/>
      <geometry>
        <box size="0.8 0.8 0.567"/>
      </geometry>
    </collision> -->
  </link>
  <link name="box_origin">
  </link>
  <joint name="iiwa_torso_base_joint" type="fixed">
    <parent link="torso_origin"/>
    <child link="iiwa_torso_base"/>
    <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
  </joint>
  <joint name="iiwa_torso_left_joint" type="fixed">
    <parent link="iiwa_torso_base"/>
    <child link="iiwa_torso_left"/>
    <origin rpy="-1.01985341 -0.3007097 -0.70361386" xyz="0.065 0.1126 0.10074"/>
  </joint>
  <joint name="iiwa_torso_right_joint" type="fixed">
    <parent link="iiwa_torso_base"/>
    <child link="iiwa_torso_right"/>
    <origin rpy="1.01985341 -0.3007097  0.70361386" xyz="0.065 -0.1126 0.10074"/>
  </joint>
  <joint name="iiwa_torso_back_joint" type="fixed">
    <parent link="iiwa_torso_base"/>
    <child link="iiwa_torso_back"/>
    <origin rpy="1.01985341 -0.3007097 -1.39078124" xyz="-0.13 0 0.10074"/>
  </joint>
  <link name="iiwa_torso_left">
    </link>
  <link name="iiwa_torso_right">
    </link>
  <link name="iiwa_torso_back">
    </link>
  <link name="iiwa_torso_base">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="17.097"/>
      <inertia ixx="0.231426028862462" ixy="3.4651505109641E-07" ixz="9.00820316949685E-07" iyy="0.352096317978007" iyz="1.23740976421988E-05" izz="0.231751030960783"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://bh_description/meshes/iiwa_mount/iiwa_torso.stl"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8667 0.86667 0.8902 1"/>
      </material>
    </visual>
    <!--
        <collision>
        <origin
            xyz="${mount_origin}" rpy="0 0 0" />
        <geometry>
            <mesh
            filename="package://bh_description/meshes/iiwa_mount/iiwa_torso.stl" />
        </geometry>
        </collision>
        -->
  </link>
  <!--joint between {parent} and link_0-->
  <joint name="iiwa_torso_left_bh_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="iiwa_torso_left"/>
    <child link="bh_link_0"/>
  </joint>
  <link name="bh_link_0">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.1 0 0.07"/>
      <mass value="5"/>
      <inertia ixx="0.05" ixy="0" ixz="0" iyy="0.06" iyz="0" izz="0.03"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/visual/link_0.stl"/>
        <!-- <box size="0.24 0.23 0.1575"/> -->
      </geometry>
      <material name="Grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/collision/link_0.stl"/>
        <!-- <box size="0.24 0.23 0.1575"/> -->
      </geometry>
      <material name="Grey"/>
    </collision>
    <self_collision_checking>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <capsule length="0.25" radius="0.15"/>
      </geometry>
    </self_collision_checking>
  </link>
  <!-- joint between link_0 and link_1 -->
  <joint name="bh_joint_1" type="revolute">
    <parent link="bh_link_0"/>
    <child link="bh_link_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.15"/>
    <axis xyz="0 0 1"/>
    <limit effort="176" lower="-2.81870674197" upper="2.81870674197" velocity="1.71042266695"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.81870674197" soft_upper_limit="2.81870674197"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="bh_link_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0 -0.03 0.12"/>
      <mass value="3.4525"/>
      <inertia ixx="0.02183" ixy="0" ixz="0" iyy="0.007703" iyz="-0.003887" izz="0.02083"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0075"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/visual/link_1.stl"/>
        <!-- <box size="0.175 0.19 0.26"/> -->
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0075"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/collision/link_1.stl"/>
        <!-- <box size="0.175 0.19 0.26"/> -->
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="bh_joint_2" type="revolute">
    <parent link="bh_link_1"/>
    <child link="bh_link_2"/>
    <origin rpy="1.57079632679   0 3.14159265359" xyz="0 0 0.19"/>
    <axis xyz="0 0 1"/>
    <limit effort="176" lower="-1.98967534727" upper="1.98967534727" velocity="1.71042266695"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-1.98967534727" soft_upper_limit="1.98967534727"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="bh_link_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0003 0.059 0.042"/>
      <mass value="3.4821"/>
      <inertia ixx="0.02076" ixy="0" ixz="-0.003626" iyy="0.02179" iyz="0" izz="0.00779"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/visual/link_2.stl"/>
        <!-- <box size="0.175 0.24 0.19"/> -->
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/collision/link_2.stl"/>
        <!-- <box size="0.175 0.24 0.19"/> -->
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="bh_joint_3" type="revolute">
    <parent link="bh_link_2"/>
    <child link="bh_link_3"/>
    <origin rpy="1.57079632679 0 3.14159265359" xyz="0 0.21 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="110" lower="-2.81870674197" upper="2.81870674197" velocity="1.74532925199"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.81870674197" soft_upper_limit="2.81870674197"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="bh_link_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.03 0.13"/>
      <mass value="4.05623"/>
      <inertia ixx="0.03204" ixy="0" ixz="0" iyy="0.00972" iyz="0.006227" izz="0.03042"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.026"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/visual/link_3.stl"/>
        <!-- <box size="0.125 0.15 0.26"/> -->
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.026"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/collision/link_3.stl"/>
        <!-- <box size="0.125 0.15 0.26"/> -->
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="bh_joint_4" type="revolute">
    <parent link="bh_link_3"/>
    <child link="bh_link_4"/>
    <origin rpy="1.57079632679 0 0" xyz="0 0 0.19"/>
    <axis xyz="0 0 1"/>
    <limit effort="110" lower="-1.98967534727" upper="1.98967534727" velocity="2.26892802759"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-1.98967534727" soft_upper_limit="1.98967534727"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="bh_link_4">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.067 0.034"/>
      <mass value="3.4822"/>
      <inertia ixx="0.02178" ixy="0" ixz="0" iyy="0.02075" iyz="-0.003625" izz="0.007785"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/visual/link_4.stl"/>
        <!-- <box size="0.125 0.22 0.15"/> -->
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/collision/link_4.stl"/>
        <!-- <box size="0.125 0.22 0.15"/> -->
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="bh_joint_5" type="revolute">
    <parent link="bh_link_4"/>
    <child link="bh_link_5"/>
    <origin rpy="-1.57079632679 3.14159265359 0" xyz="0 0.21 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="110" lower="-2.81870674197" upper="2.81870674197" velocity="2.44346095279"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.81870674197" soft_upper_limit="2.81870674197"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="bh_link_5">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0001 0.021 0.076"/>
      <mass value="2.1633"/>
      <inertia ixx="0.01287" ixy="0" ixz="0" iyy="0.005708" iyz="-0.003946" izz="0.01112"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.026"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/visual/link_5.stl"/>
        <!-- <box size="0.125 0.15 0.24"/> -->
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.026"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/collision/link_5.stl"/>
        <!-- <box size="0.125 0.15 0.24"/> -->
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="bh_joint_6" type="revolute">
    <parent link="bh_link_5"/>
    <child link="bh_link_6"/>
    <origin rpy="1.57079632679 0 0" xyz="0 0.06070 0.19"/>
    <axis xyz="0 0 1"/>
    <limit effort="40" lower="-1.98967534727" upper="1.98967534727" velocity="3.14159265359"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-1.98967534727" soft_upper_limit="1.98967534727"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="bh_link_6">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.0006 0.0004"/>
      <mass value="2.3466"/>
      <inertia ixx="0.006509" ixy="0" ixz="0" iyy="0.006259" iyz="0.00031891" izz="0.004527"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/visual/link_6.stl"/>
        <!-- <box size="0.15 0.15 0.15"/> -->
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/collision/link_6.stl"/>
        <!-- <box size="0.15 0.15 0.15"/> -->
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="bh_joint_7" type="revolute">
    <parent link="bh_link_6"/>
    <child link="bh_link_7"/>
    <origin rpy="-1.57079632679 3.14159265359 0" xyz="0 0.081 0.06070"/>
    <axis xyz="0 0 1"/>
    <limit effort="40" lower="-2.90160988144" upper="2.90160988144" velocity="3.14159265359"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.90160988144" soft_upper_limit="2.90160988144"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="bh_link_7">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.02"/>
      <mass value="3.129"/>
      <inertia ixx="0.01464" ixy="0.0005912" ixz="0" iyy="0.01465" iyz="0" izz="0.002872"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.0005"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/visual/link_7.stl"/>
        <!-- <box size="0.13 0.13 0.06"/> -->
      </geometry>
      <material name="Grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.0005"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/iiwa7/collision/link_7.stl"/>
        <!-- <box size="0.13 0.13 0.06"/> -->
      </geometry>
      <material name="Grey"/>
    </collision>
  </link>
  <joint name="bh_joint_ee" type="fixed">
    <parent link="bh_link_7"/>
    <child link="bh_link_ee"/>
    <origin rpy="0 0 0" xyz="0 0 0.045"/>
  </joint>
  <link name="bh_link_ee">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1e-6"/>
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
    </inertial>
  </link>
  <joint name="ee_peg_base_joint" type="fixed">
    <parent link="bh_link_ee"/>
    <child link="peg_base"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>

  <joint name="peg_end_joint" type="fixed">
    <parent link="peg_base"/>
    <child link="peg"/>
    <origin rpy="0 0 0" xyz="0 0 0.1"/>
  </joint>

  <link name="peg"/>

  <link name="peg_base">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <!-- Rough guess -->
      <mass value="0.3"/>
        <inertia ixx="0.000312499875" ixy="0.0" ixz="0.0" iyy="0.000312499875" iyz="0.0" izz="0.00012499995"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://bh_description/meshes/peg_and_hole/peg_30mm_hr.stl" scale="0.001 0.001 0.001" />
        <!-- <box size="0.05 0.05 0.2"/> -->
      </geometry>
      <material name="Black"/>
      <!-- <material name="indigo">
        <color rgba="0.25 0.0 0.5 1"/>
      </material> -->
    </visual>
    <collision>
      <origin
          xyz="0 0 0" rpy="0 0 0" />
      <geometry>
          <mesh filename="package://bh_description/meshes/peg_and_hole/peg_30mm_hr.stl" scale="0.001 0.001 0.001" />
          <!-- <box size="0.05 0.05 0.2"/> -->
      </geometry>
    </collision>
  </link>

  <!-- Enable the Joint Feedback -->
  <gazebo reference="bh_joint_7">
    <provideFeedback>true</provideFeedback>
  </gazebo>
  <!-- The ft_sensor plugin -->
  <gazebo>
    <plugin filename="libgazebo_ros_ft_sensor.so" name="ft_sensor">
      <updateRate>50.0</updateRate>
      <topicName>/bh/state/CartesianWrench</topicName>
      <jointName>bh_joint_7</jointName>
    </plugin>
  </gazebo>
  <!-- Load Gazebo lib and set the robot namespace -->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>/bh</robotNamespace>
      <robotSimType>iiwa_gazebo/GravityCompensationHWSim</robotSimType>
    </plugin>
  </gazebo>
  <!-- Link0 -->
  <gazebo reference="bh_link_0">
    <material>Gazebo/Grey</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>
  <!-- Link1 -->
  <gazebo reference="bh_link_1">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>
  <!-- Link2 -->
  <gazebo reference="bh_link_2">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>
  <!-- Link3 -->
  <gazebo reference="bh_link_3">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>
  <!-- Link4 -->
  <gazebo reference="bh_link_4">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>
  <!-- Link5 -->
  <gazebo reference="bh_link_5">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>
  <!-- Link6 -->
  <gazebo reference="bh_link_6">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>
  <!-- Link7 -->
  <gazebo reference="bh_link_7">
    <material>Gazebo/Grey</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>
  <transmission name="bh_tran_1">
    <robotNamespace>/bh</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="bh_joint_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="bh_motor_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="bh_tran_2">
    <robotNamespace>/bh</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="bh_joint_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="bh_motor_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="bh_tran_3">
    <robotNamespace>/bh</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="bh_joint_3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="bh_motor_3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="bh_tran_4">
    <robotNamespace>/bh</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="bh_joint_4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="bh_motor_4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="bh_tran_5">
    <robotNamespace>/bh</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="bh_joint_5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="bh_motor_5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="bh_tran_6">
    <robotNamespace>/bh</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="bh_joint_6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="bh_motor_6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="bh_tran_7">
    <robotNamespace>/bh</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="bh_joint_7">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="bh_motor_7">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!--<primesense_camera name="realsense_$(arg left)" parent="$(arg left)_link_ee" joint_type="$(arg camera_joint_type)">
      <origin xyz="$(arg camera_xyz)" rpy="$(arg camera_rpy)"/>
    </primesense_camera>-->
</robot>

